src/main/cpp/apple_fsnotifier.cpp

```diff
@@ -78,14 +78,14 @@ static void callback(ConstFSEventStreamRef streamRef,
             } else {
                 type = FILE_EVENT_CREATED;
             }
+        } else if (IS_SET(flags, kFSEventStreamEventFlagItemModified)) {
+            type = FILE_EVENT_MODIFIED;
         } else if (IS_ANY_SET(flags,
                 kFSEventStreamEventFlagItemInodeMetaMod // file locked
```

Comment:
Let's do the check for metadata changes after the modified/created/removed changes. I think those flags trump the metadata changes.